component/src/main/java/org/wso2/extension/siddhi/io/feed/source/FeedListener.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.feed.source;
+
+import org.apache.abdera.Abdera;
+import org.apache.abdera.model.Document;
+import org.apache.abdera.model.Entry;
+import org.apache.abdera.model.Feed;
+import org.apache.axiom.om.OMElement;
+import org.apache.log4j.Logger;
+import org.wso2.extension.siddhi.io.feed.utils.Constants;
+import org.wso2.extension.siddhi.io.feed.utils.EntryUtils;
+import org.wso2.siddhi.core.stream.input.source.SourceEventListener;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This Runnable class consume Atom and RSS doc types and push entries in to Siddhi.
+ */
+public class FeedListener implements Runnable {
+    private Logger log = Logger.getLogger(FeedListener.class);
+    private boolean isPause = false;
+    private ReentrantLock lock = new ReentrantLock();
+    private Condition condition = lock.newCondition();
+    private SourceEventListener sourceEventListener;
+    private URL url;
+    private Abdera abdera;
+    private String type;
+
+    public FeedListener(SourceEventListener sourceEventListener, URL url, String type) {
+        this.sourceEventListener = sourceEventListener;
+        this.url = url;
+        abdera = new Abdera();
+        this.type = type;
+    }
+
+    @Override
+    public void run() {
+        InputStream inputStream = null;
+        try {
+            inputStream = url.openStream();
+            if (type.equals(Constants.ATOM)) { // Atom Consuming
```

Comment:
If these cases are not case sensitive, use ignore case while checking the equality